File: CST-105/BirthdayGuesser/src/BirthdayGuesser.java

public class BirthdayGuesser
{
	public static final int GridSize = 4;
	private static final int MaxGuesses = 5;
	private int currDayGrid = 0;
	
	private int[][][] dates =
		{
			{
				{1, 3, 5, 7},
				{9, 11, 13, 15},
				{17, 19, 21, 23},
				{25, 27, 29, 31}
			},
			{
				{2, 3, 6, 7},
				{10, 11, 14, 15},
				{18, 19, 22, 23},
				{26, 27, 30, 31}
			},
			{
				{4, 5, 6, 7},
				{12, 13, 14, 15},
				{20, 21, 22, 23},
				{28, 29, 30, 31}
			},
			{
				{8, 9, 10, 11},
				{12, 13, 14, 15},
				{24, 25, 26, 27},
				{28, 29, 30, 31}
			},
			{
				{16, 17, 18, 19},
				{20, 21, 22, 23},
				{24, 25, 26, 27},
				{28, 29, 30, 31}
			}
		 };

	public BirthdayGuesser()
	{
	}
	
	public int[][] GetCurrentDayGrid()
	{
		return dates[currDayGrid];
	}
	
	public int GetAddDays()
	{
		return dates[currDayGrid][0][0];
	}
	
	public boolean NextGrid()
	{
		if (currDayGrid < MaxGuesses - 1)
		{
			currDayGrid ++;
			return true;
		}
		
		return false;
	}
}

File: CST-210/BirthdayGuesser/BirthdayGuesser/BirthdayGuesser.cpp
// BirthdayGuesser.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

#include "BirthdaySolver.h"

int main()
{
    BirthdaySolver guesser;
    int day = 0;

    do
    {
        auto currGrid = guesser.GetCurrentDayGrid();
        for (auto row = 0; row < GridSize; row ++)
        {
            for (auto column = 0; column < GridSize; column ++)
            {
                cout << setw(4) << currGrid[row][column];
            }

            cout << endl;
        }

        cout << endl;
        cout << "Is your birthday in this set? (y/n) ";
        string answer;
        getline(cin, answer);
        if (tolower(answer[0]) == 'y')
        {
            day += guesser.GetAddDays();
        }
    } while (guesser.NextGrid());

    cout << "Your birthday is " << day << endl;
}
